spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/StructAccessor.java

```diff
@@ -79,17 +80,20 @@
 		this.columnNamesIndex = indexColumnNames();
 	}
 
-	public Object getSingleValue(String colName) {
+	Object getSingleValue(String colName) {
 		Type colType = this.struct.getColumnType(colName);
 		Type.Code code = colType.getCode();
 		Class sourceType = code.equals(Type.Code.ARRAY)
 				? this.spannerTypeMapper.getArrayJavaClassFor(colType.getArrayElementType().getCode())
 				: this.spannerTypeMapper.getSimpleJavaClassFor(code);
 		BiFunction readFunction = singleItemReadMethodMapping.get(sourceType);
+		if (readFunction == null) {
```

Comment:
when can this realistically happen? (i.e. not in _test_ code)

Reply:
I'm not sure, but I think it's good to return null here. In the reader we throw the exception if at the end the read value was null.

Reply:
I got this one to come up by not having a needed custom converter supplied. The readfunction in that was null, and that caused a nullpointerexception.

Reply:
This doesn't seem to have to do much with converters. The `singleItemReadMethodMapping` keyset should cover the column types...

Reply:
I lost my original example, but I remember it came up when the POJO had a singular property (like int) and the corresponding column was ARRAY<STRUCT>. This causes `sourceType` to be `Struct[].class`, which isn't in `singleItemReadMethodMapping`, so readFunction is null. 

If we return null then we get the exception later saying "int couldn't be read from Struct[ ] " which makes sense to me.  It looks like this would happen if the column was some other types like "ARRAY<BYTES>" and the POJO property was singular, too. 

Alternatively we could explicitly check for singular-property vs. array-column situation, but it seems like  we naturally detect these situations based on the way we have the type/function maps set up because we will just fail to find a usable function (which  feels pretty elegant).

Reply:
1. we need to have an example in unit test form - otherwise someone might just remove this piece of code and wouldn't know that a valid case was broken
2. StructAccessor accesses a Struct. This method gets a colName. There are no converters in the picture. This must mean that there is a column type that we are not capturing in the `singleItemReadMethodMapping`. And I guess that could be the `Struct[]` or `Struct`? The question is how do we want to handle on this abstraction level these missing column types.

Reply:
I "misspoke" about the converter. Please ignore that brief mention from earlier. The case I wanted to illustrate is the longer last comment I had.

Reply:
can you just simply please write a realistic testcase that exercises this path? Then we can figure out the right design for it.

Reply:
I actually did in a commit earlier today.

take a look at 02fdef6

Reply:
Wow that's a really subtle find! I would never have expected the default conversion service we're using can convert from single-item lists to a single item!

I think it's a good idea to add `.put(Struct[].class, (struct, col) -> null)` as a safe-guard. This situation is looks the same for ByteArray[].class/getByteArrayList, Timestamp[].class/getTimestampList, etc. , and we ought to add the null-returning anon function for those too, right?

Reply:
As discussed, just adding a comment will suffice. Thanks.